lib/streamlit/elements/vega_charts.py

```diff
@@ -74,6 +90,42 @@
 }
 
 
+class VegaLiteState(TypedDict, total=False):
+    """
+    A dictionary representing the current selection state of the VegaLite chart.
+    Attributes
+    ----------
+    select : AttributeDictionary
+        The state of the `on_select` event.
+    """
+
+    select: AttributeDictionary
+
+
+@dataclass
+class VegaLiteStateSerde:
+    """VegaLiteStateSerde is used to serialize and deserialize the VegaLite Chart state."""
+
+    def deserialize(self, ui_value: str | None, widget_id: str = "") -> VegaLiteState:
+        empty_selection_state: VegaLiteState = {
+            "select": AttributeDictionary(),
+        }
+
+        selection_state = (
+            empty_selection_state
+            if ui_value is None
+            else cast(VegaLiteState, AttributeDictionary(json.loads(ui_value)))
+        )
```

Comment:
this could lead to a cast exception, right? Should we handle this gracefully? At least we should add a docstring comment

Reply:
afaik `cast` is only relevant during type checking, it doesn't do anything at runtime. So, I don't think there is a possibility of a cast exception

Reply:
oh yeah you are right according [to the docs](https://docs.python.org/3/library/typing.html#typing.cast)